dbt/adapters/databricks/impl.py

```diff
@@ -101,9 +101,19 @@ class DatabricksConfig(AdapterConfig):
     buckets: Optional[int] = None
     options: Optional[Dict[str, str]] = None
     merge_update_columns: Optional[str] = None
+    merge_exclude_columns: Optional[str] = None
     databricks_tags: Optional[Dict[str, str]] = None
     tblproperties: Optional[Dict[str, str]] = None
     zorder: Optional[Union[List[str], str]] = None
+    skip_non_matched_step: Optional[bool] = None
+    skip_matched_step: Optional[bool] = None
+    matched_condition: Optional[str] = None
+    not_matched_condition: Optional[str] = None
+    not_matched_by_source_action: Optional[str] = None
+    not_matched_by_source_condition: Optional[str] = None
+    target_alias: Optional[str] = None
```

Comment:
I wonder if these should be nested in a 'merge_options' config?  Mostly thinking about this because target_alias and source_alias might have meaning in other scenarios, but we want users to know that this is a mechanism for tailoring the merge SQL.

Reply:
That's possible, but we have to possibly do the same with some of the old configuration parameters.

So right now it looks like this

```bash
└── config
    ├── target_alias
    ├── source_alias
    ├── predicates
    ├── dest_columns
    ├── source_columns
    ├── merge_update_columns
    ├── merge_exclude_columns
    ├── merge_with_schema_evolution
    ├── on_schema_change
    ├── update_columns
    ├── skip_matched_step
    ├── skip_not_matched_step
    ├── matched_condition
    ├── not_matched_condition
    ├── not_matched_by_source_action
    └── not_matched_by_source_condition
```

If I move only added options to the `merge_options` subdict it will look like this

```bash
└── config
    ├── predicates
    ├── dest_columns
    ├── source_columns
    ├── merge_update_columns
    ├── merge_exclude_columns
    ├── on_schema_change
    ├── update_columns
    └── merge_options/
        ├── target_alias
        ├── source_alias
        ├── merge_with_schema_evolution
        ├── skip_matched_step
        ├── skip_not_matched_step
        ├── matched_condition
        ├── not_matched_condition
        ├── not_matched_by_source_action
        └── not_matched_by_source_condition
```

where at least for me the rest of the options are misplaced, as they are also `merge_options`.

## Options

Though I don't mind to change the code, I think it is a quite important design solution. So let me outline my opinion for the options that I see here.

With a **flat config map** there's a hard requirement to have all options uniquely identified to exclude misinterpretation. Also the documentation will be absolutely required.
However, If some config parameter will be misused on `dbt-databricks` development side - it will likely fail the tests. If not - it should be safe to use it.
Same time flat map simplifies the structure and kind of make it more straightforward for users, especially vs the case when some parameters should be nested and some not. Also, if some parameters should be used in several contexts - it is okay to use existing config key.

With a **merge options subdict** we group some of the options highlighting the usage context. But, if some identical parameter (we are using SQL, where many clauses are also coincide between statements) should be used for a different context - we will have to create another subdict or introduce a root `config` key. 
Also, I think in this case we should migrate all existing parameters to `merge_options` as well, which will create the need for deprecation process.

So in the context of the current state of the `config` and proposed extension I perceive the variant with `merge_options` subdict as an unnecessary complexity. 

Of course, I will fix everything to the state that you think is better for the future of the product.

Reply:
There's a middle way. We can keep flat map, but highlight some critical config keys that they are exclusively for the merge context. By adding a `merge_` prefix, for instance.

```bash
└── config
    ├── merge_target_alias
    ├── merge_source_alias
    ├── predicates
    ├── dest_columns
    ├── merge_source_columns
    ├── merge_update_columns
    ├── merge_exclude_columns
   ...
```

Bad thing is that still some old parameters seem to have identical problem (`predicates`, `dest_columns`) and if we have to deprecate them in favor of a new name - then we can probably do this later 😄

Reply:
So, you can't see it here, because I'm just reading it out of extras, but I'm already using nesting; I'll accept flat for now, and we'll see if any conflicts come up later.